make.js

```diff
@@ -204,6 +204,9 @@ CLI.serverBuild = function(/** @type {{ task: string }} */ argv) {
     // Need to validate generated tasks first
     const makeOptions = fileToJson(makeOptionsPath);
 
+    if (!callGenTaskDuringBuild) {
+        generateTasksOnServerBuild();
+    }
```

Comment:
generateTasksOnServerBuild and processGeneratedTasks are very similar (both call buildConfigGenerator).  Can we combine them?

Reply:
@merlynomsft, great idea, [processGeneratedTasks ](https://github.com/microsoft/azure-pipelines-tasks/blob/master/make-util.js#L1866) doing exactly what we need, We just need to pass writeUpdates: true to regenerate tasks [here](https://github.com/microsoft/azure-pipelines-tasks/blob/master/make.js#L207) during the serverBuild, I think we could just remove callGenTaskDuringBuild variable and always pass true, because we started re-generate such tasks no matter where we build them(ci or locally)?

Reply:
Updated